modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/CommonScriptMessageContext.java

```diff
@@ -0,0 +1,846 @@
+            }
+            json.append("}");
+        } else if (obj instanceof NativeArray) {
+            json.append("[");
+            NativeArray o = (NativeArray) obj;
+            Object[] ids = o.getIds();
+            boolean first = true;
+            for (Object id : ids) {
+                Object value = o.get((Integer) id, o);
+                if (!first) {
+                    json.append(", ");
+                } else {
+                    first = false;
+                }
+                json.append(serializeJSON(value));
+            }
+            json.append("]");
+        } else if (obj instanceof Object[]) {
+            json.append("[");
+            boolean first = true;
+            for (Object value : (Object[]) obj) {
+                if (!first) {
+                    json.append(", ");
+                } else {
+                    first = false;
+                }
+                json.append(serializeJSON(value));
+            }
+            json.append("]");
+
+        } else if (obj instanceof String) {
+            json.append("\"").append(obj.toString()).append("\"");
+        } else if (obj instanceof Integer ||
+                obj instanceof Long ||
+                obj instanceof Float ||
+                obj instanceof Double ||
+                obj instanceof Short ||
+                obj instanceof BigInteger ||
+                obj instanceof BigDecimal ||
+                obj instanceof Boolean) {
+            json.append(obj.toString());
+        } else {
+            json.append("{}");
+        }
+        return json.toString();
+    }
+
+    @Override
+    public void setPayloadJSON(Object jsonPayload) throws ScriptException {
+        org.apache.axis2.context.MessageContext messageContext;
+        messageContext = ((Axis2MessageContext) mc).getAxis2MessageContext();
+
+        byte[] json = {'{', '}'};
+        if (jsonPayload instanceof String) {
+            json = jsonPayload.toString().getBytes();
+        } else if (jsonPayload != null) {
+            try {
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+                serializeJson(jsonPayload, out);
+                json = out.toByteArray();
+            } catch (IOException e) {
+                logger.error("#setPayloadJSON. Could not retrieve bytes from JSON object. Error>>> ", e);
+            }
+        }
+        // save this JSON object as the new payload.
+        try {
+            JsonUtil.getNewJsonPayload(messageContext, json, 0, json.length, true, true);
+        } catch (AxisFault axisFault) {
+            throw new ScriptException(axisFault);
+        }
+        //JsonUtil.setContentType(messageContext);
+        Object jsonObject = scriptEngine.eval(JsonUtil.newJavaScriptSourceReader(messageContext));
+        setJsonObject(mc, jsonObject);
+    }
+
+    private void serializeJson(Object obj, OutputStream out) throws IOException {
```

Comment:
Shall we move this also to util class? may be useful to other features
better to move synapse-commons